packages/xsnap/src/avaAssertXS.js

```diff
@@ -0,0 +1,349 @@
+    },
+  });
+}
+
+/** @type { Harness | null } */
+let theHarness = null; // ISSUE: ambient
+
+/**
+ * @param {(msg: TapMessage) => void} send
+ *
+ * @typedef { ReturnType<typeof createHarness>} Harness
+ */
+function createHarness(send) {
+  let testNum = 0;
+  let passCount = 0;
+  /** @type {((ot: { context: Object }) => Promise<void>)[]} */
+  let beforeHooks = [];
+  /** @type {(() => Promise<void>)[]} */
+  let suitesToRun = [];
+  const context = {};
+
+  /**
+   * @returns { Summary }
+   * @typedef {import('./avaXS').Summary} Summary
+   */
+  function summary() {
+    return {
+      pass: passCount,
+      fail: testNum - passCount,
+      total: testNum,
+    };
+  }
+
+  const it = freeze({
+    send,
+    get context() {
+      return context;
+    },
+    /** @type {(label: string, hook: () => Promise<void>) => void } */
+    before(_label, hook) {
+      beforeHooks.push(hook);
+    },
+    /** @type { (ok: boolean) => number } */
+    finish(ok) {
+      testNum += 1;
+      if (ok) {
+        passCount += 1;
+      }
+      return testNum;
+    },
+    /** @type { (thunk: () => Promise<void>) => Promise<void> } */
+    async defer(thunk) {
+      suitesToRun.push(thunk);
+    },
+    summary,
+    async result() {
+      for await (const hook of beforeHooks) {
+        await hook({ context });
+      }
+      beforeHooks = [];
+      for await (const suite of suitesToRun) {
+        await suite();
+      }
+      suitesToRun = [];
+
+      return summary();
+    },
+  });
+
+  if (!theHarness) {
+    theHarness = it;
+  }
+  return it;
+}
+
+/**
+ * @param {*} exc
+ * @param {Expectation} expectation
+ * @returns {boolean}
+ * @typedef {{ instanceOf: Function } | { message: string | RegExp }=} Expectation
+ */
+function checkExpectation(exc, expectation) {
+  if (!expectation) return true;
+  if ('instanceOf' in expectation) return exc instanceof expectation.instanceOf;
+  if ('message' in expectation) {
+    const { message } = expectation;
+    return typeof message === 'string'
+      ? message === exc.message
+      : message.test(exc.message);
+  }
+  throw Error(`not implemented: ${JSON.stringify(expectation)}`);
+}
+
+/**
+ * Emulate ava assertion API
+ *
+ * ref https://github.com/avajs/ava/blob/main/docs/03-assertions.md
+ *
+ * @param {Harness} htest
+ * @param {TapFormat} out
+ *
+ * @typedef {ReturnType<typeof makeTester>} Tester
+ */
+function makeTester(htest, out) {
+  /** @type {number?} */
+  let pending;
+
+  /** @type {(result: boolean, info?: string) => void} */
+  function assert(result, info) {
+    if (typeof pending === 'number') {
+      pending -= 1;
+    }
+    const testNum = htest.finish(result);
+    if (result) {
+      out.ok(testNum, info);
+    } else {
+      out.notOk(testNum, info);
+    }
+  }
+
+  function truthy(/** @type {unknown} */ value, msg = 'should be truthy') {
+    assert(!!value, msg);
+  }
+
+  const t = freeze({
+    plan(/** @type {number} */ count) {
```

Comment:
Then why not write it the way we write other parameter declarations?

```suggestion
    /**
     * @param {number} count
     */
    plan(count) {
```

Reply:
because it would double the size of the 3-line method. The amount of code that fits on the screen matters, no?